ts/IterableInbox.tsx

```diff
@@ -0,0 +1,318 @@
+   InboxImpressionRowInfo
+} from '.'
+
+import { Iterable } from './Iterable'
+
+import IterableInboxMessageList from './IterableInboxMessageList'
+import IterableInboxMessageDisplay from './IterableInboxMessageDisplay'
+import IterableInboxDataModel from './IterableInboxDataModel'
+import InboxRowViewModel from './InboxRowViewModel'
+
+import { useIsFocused } from '@react-navigation/native'
+
+const RNIterableAPI = NativeModules.RNIterableAPI
+const RNEventEmitter = new NativeEventEmitter(RNIterableAPI)
+
+type inboxProps = {
+   returnToInboxTrigger?: boolean,
+   messageListItemLayout?: Function,
+   customizations?: IterableInboxCustomizations,
+   tabBarHeight?: number,
+   tabBarPadding?: number,
+   safeAreaMode?: boolean
+}
+
+const IterableInbox = ({
+   returnToInboxTrigger = true,
+   messageListItemLayout = () => { return null },
+   customizations = {} as IterableInboxCustomizations,
+   tabBarHeight = 80,
+   tabBarPadding = 20,
+   safeAreaMode = true
+}: inboxProps) => {
+   const defaultInboxTitle = "Inbox"
+   const inboxDataModel = new IterableInboxDataModel()
+
+   let { height, width, isPortrait } = useDeviceOrientation()
+   const appState = useAppStateListener()
+   const isFocused = useIsFocused()
+
+   const [selectedRowViewModelIdx, setSelectedRowViewModelIdx] = useState<number>(0)
+   const [rowViewModels, setRowViewModels] = useState<InboxRowViewModel[]>([])
+   const [loading, setLoading] = useState<boolean>(true)
+   const [animatedValue] = useState<any>(new Animated.Value(0))
+   const [isMessageDisplay, setIsMessageDisplay] = useState<boolean>(false)
+   
+   const [visibleMessageImpressions, setVisibleMessageImpressions] = useState<InboxImpressionRowInfo[]>([])
+
+   const styles = StyleSheet.create({
+      loadingScreen: {
+         height: '100%',
+         backgroundColor: 'whitesmoke'
+      },
+   
+      container: {
+         flex: 1,
+         flexDirection: 'row',
+         alignItems: 'center',
+         justifyContent: 'flex-start',
+         height: '100%',
+         width: 2 * width,
+         paddingBottom: 0,
+         paddingLeft: 0,
+         paddingRight: 0
+      },
+   
+      messageListContainer: {
+         height: '100%',
+         width: width,
+         flexDirection: 'column',
+         justifyContent: 'flex-start',
+      },
+   
+      headline: {
+         fontWeight: 'bold',
+         fontSize: 40,
+         width: '100%',
+         height: 60,
+         marginTop: 0,
+         paddingTop: 10,
+         paddingBottom: 10,
+         paddingLeft: 30,
+         backgroundColor: 'whitesmoke'
+      }
+   })
+
+   let {
+      loadingScreen,
+      container,
+      headline,
+      messageListContainer
+   } = styles
+
+   const navTitleHeight = headline.height + headline.paddingTop + headline.paddingBottom
+   headline = { ...headline, height: Platform.OS === "android" ? 70 : 60 }
+   headline = (!isPortrait) ? { ...headline, paddingLeft: 70 } : headline
+
+   useEffect(() => {
```

Comment:
Let's leave a comment on each `useEffect` for future reference.

Reply:
Adding comments.